### ISSUE TITLE
Troubleshooting Azure CycleCloud Deployment Steps

### ISSUE THREAD

**Issue by BiologyGeek** [microsoft/AzureTRE#3933]
Hello team,

I wanted to deploy Azure CycleCloud and have referred to [this document](https://microsoft.github.io/AzureTRE/latest/tre-templates/shared-services/cyclecloud/). However, I encountered some confusion while attempting to deploy this service. After running the relevant command to approve the license terms, I tried the following commands using a dev container but without success:

```cmd
$ make cyclecloud
$ make shared_service_bundle BUNDLE=cyclecloud
$ make workspace_service_bundle BUNDLE=cyclecloud

$ cd AzureTRE/templates/shared_services/cyclecloud
➜ .../templates/shared_services/cyclecloud/terraform $ bash deploy.sh
```

The document states:
> The CycleCloud shared service template needs registering with the TRE as per <../../tre-admins/registering-templates/> The templates can be found at templates/shared_services/cyclecloud.

I am wondering which method is correct for registering this service. What went wrong? Is this method correct, or did I miss something?

Also, by opening this URL (https://cyclecloud-{My-Own-TRE_ID}.{My-Selected-LOCATION}.cloudapp.azure.com/) in the web browser of a virtual machine, the browser displays the 'DNS_PROBE_FINISHED_NXDOMAIN' error.

**Comment by Danny-Cooke-CK** [comment]
Hi. I would always default to these instructions for any bundle. https://microsoft.github.io/AzureTRE/latest/tre-admins/registering-templates/
What version of AzureTRE are you using ?

**Comment by tim-allen-ck** [comment]
hi @BiologyGeek Im looking into the issue. 
I believe the problem is these 4 lines in the porter.yaml file in the cyclecloud folder
```yaml
arm_client_id: ${ bundle.credentials.azure_client_id }
arm_client_secret: ${ bundle.credentials.azure_client_secret }
arm_tenant_id: ${ bundle.credentials.azure_tenant_id }
arm_use_msi: ${ bundle.parameters.arm_use_msi }
```
they need removing, then try to build and publish the service.

**Comment by tim-allen-ck** [comment]
Hi @BiologyGeek 
According to the docs, you'll need to head to the public url of the cyclecloud instance. 
`https://cyclecloud-{TRE_ID}.{LOCATION}.cloudapp.azure.com/` 
https://microsoft.github.io/AzureTRE/v0.16.0/tre-templates/shared-services/cyclecloud/

**Comment by tim-allen-ck** [comment]
Regarding 
> **Question:** By running `$ make bundle-publish` and `$ make bundle-register`, will it override the impact of the `$ make shared_service_bundle BUNDLE=cyclecloud` commands that were run previously, or does something need to be manually unregistered in some way?

It should only override if you've edited the version of the bundle in the porter.yaml

**Comment by tim-allen-ck** [comment]
Hi @BiologyGeek, I'll have a look at this next week and get back to you.

**Comment by tim-allen-ck** [comment]
hi @BiologyGeek, sorry it's taken a while.
I think I've fixed the management API issue, looks like it needs an extra rule in the firewall to allow 80 and 443 access to the management.azure.com fqdn
<img width="1068" alt="image" src="https://github.com/microsoft/AzureTRE/assets/139883291/5d8d8e72-2437-44a8-9310-a01939e649ac">
also as per the docs I've been testing this in the admin VM which I connected to via Bastion and that seems to work using the private dns address and that seems to be working 
<img width="696" alt="image" src="https://github.com/microsoft/AzureTRE/assets/139883291/293b2797-c357-4c5e-a89d-83826757866f">

**Comment by marrobi** [comment]
Fixed  in #4050
